{% from "macros.html" import not_none, parse_cite_as, show_authors %}
{% extends 'base.html' %}
{% block content %}

<div class="row" style="margin-right: 0; margin-left: 0;">
  <div class="col-lg-5 mx-auto">
    <!-- <h3 class="media-heading">About</h3> -->
    <h3>Kipoi: Model zoo for genomics</h3>
    <p>
      Kipoi is a repository for predictive models in genomics. It
      provides a unified framework to archive, share, access, use and
      build on models developed by the community. Kipoi models come
      with code to preprocess and load input data in major file
      formats, which facilitates easy application to new datasets and
      creation of new derived models.
    </p>

    <h4>What can I do with Kipoi?</h4>

    <p>
      <ul>
	<li><a href="../groups/">Explore</a> hundreds of trained models 
	  for genomics and apply them to new data in few lines of
	  <ul>
	    <li><a href='../docs/tutorials/python-api/'>python</a></li>
	    <li><a href='../docs/tutorials/R-api/'>R</a></li>
	    <li>or via the <a href='../docs/using/01_Getting_started/#command-line-interface-quick-start'>command line</a>.</li>
	  </ul>
	</li>
	<li>Fine-tune an existing model on a new dataset. See our 
	  <a href='../docs/tutorials/python-api/#re-train-the-keras-model'>notebook</a> 
	  and the Keras fine-tuning 
	  <a href='https://gist.github.com/fchollet/7eb39b44eb9e16e59632d25fb3119975'>example</a>.</li>
	<li>Compose a new model using existing Kipoi models as building blocks. See 
	  <a href='../docs/tutorials/composing_models/'>how</a>.</li>
	<li>Contribute models and make them accessible for others. See 
	  <a href='../docs/contributing/01_Getting_started/'>how</a>.</li>
	<li>Use and develop routines for model post-processing and interpretation. Available routines:
	  <ul>
	    <li>Scoring the impact of genetic variants on molecular phenotypes. See 
	      <a href='../veff-docs/'>docs</a> and 
	      <a href='https://github.com/kipoi/examples/tree/master/2-score_variants'>tutorial</a>.</li>
	  </ul>
	</li>
      </ul>
    </p>
    
    <p>
      Popular machine learning frameworks are easily integrated into Kipoi. So far we have integrated:
      <ul>
	<li><a href='https://www.tensorflow.org/'>Tensorflow</a></li>
	<li><a href='http://pytorch.org/'>Pytorch</a></li>
	<li><a href='https://keras.io/'>Keras</a></li>
	<li><a href='http://scikit-learn.org/'>Scikit-learn</a></li>
      </ul>
      Models developed using other frameworks or custom python code can 
      also be submitted to Kipoi. See <a href='../docs/contributing/05_Writing_model.py/'>how</a>.
    </p>

    <h4>Great! How do I get started?</h4>
    <p>
      <ul>
	<li><a href="../groups/">Explore</a> Kipoi models</li>
	<li><a href='../docs/using/01_Getting_started/'>Install</a> Kipoi</li>
	<li><a href='../docs/tutorials/python-api/'>Access and use</a> a Kipoi model</li>
	<li>Follow the <a href='https://github.com/kipoi/examples'>use-case tutorials</a></li>
	<li>Adopt <a href='https://github.com/kipoi/kipoi/tree/master/notebooks'>example notebooks</a> to your own needs </li>
	<li><a href='../docs/contributing/01_Getting_started/'>Contribute</a> a model</li>
	<li><a href='https://github.com/kipoi/kipoi/issues'>Report</a> issues and give feedback</li>
	<li><a href='https://github.com/kipoi/models/issues/55'>Join</a> the development team</li>
      </ul>
    </p>
    

    <h4>Why do we need a model zoo for genomics?</h4>
    <p>
      Genomics uses experimental and computational methods to
      understand the structure, function and evolution of biomolecules
      and processes in health and disease through the lens of the
      genome. Revolutions in DNA sequencing and associated
      high-throughput assays has led to an explosion of genomic and
      molecular profiling datasets. Computational models are now
      routinely used to assimilate and derive predictions from these
      datasets.
    </p>
    <p>
      While public databases have been developed for easy
      storage and access to genomic data, there is a lack of analogous
      repositories for computational models in genomics. Models are
      implemented in various programming languages and machine
      learning frameworks, stored in diverse formats and made
      available through different channels, such as code repositories
      and supplementary material of articles. Even with the
      availability of reliable code, replicating large models trained
      on large datasets can be challenging. The lack of a unified
      model repository makes it difficult to reproduce results, apply
      models to new data, systematically compare models and
      efficiently build on existing ones. Borrowing ideas from model
      zoos introduced in other application domains, we developed Kipoi
      with the primary goal of lowering the entry barrier to modeling
      in genomics.
    </p>

    <h4>About the team</h4>

    <p>
      Kipoi was first released on 13th March 2018. The founding team:

      <ul>
	<li>&#381;iga Avsec, Jun Cheng and Julien Gagneur, Technical University of Munich - 
	  <a href='https://www.gagneurlab.in.tum.de'>Lab website</a></li>
	<li>Roman Kreuzhuber, Lara Urban and Oliver Stegle, European Bioinformatics Institute - 
	  <a href='https://www.ebi.ac.uk/research/stegle'>Lab website</a>
	</li>
	<li>Johnny Israeli, Avanti Shrikumar, Chuan Foo and Anshul Kundaje, Stanford University - 
	  <a href='https://sites.google.com/site/anshulkundaje/'>Lab website</a>
	</li>
      </ul>
    </p>
  </div>
</div>
{% endblock %}
