{% from "macros.html" import not_none, parse_cite_as, show_authors %}
{% extends 'base.html' %}
{% block content %}

<div class="row" style="margin-right: 0; margin-left: 0;">
  <div class="col-lg-9 mx-auto">
    <div class="row mb-4">
      <div class="col media kipoi-title-box d-none d-sm-flex">
        <div class="media-left">
            <img src="{{ url_for('static', filename='img/kipoi_small.png')}}" class="media-object" alt="Kipoi logo" height="80px">
        </div>
        <div class="media-body ">
          <h3 class="media-heading">: Model zoo for genomics</h3>
        </div>
      </div>
      <div class="col media kipoi-title-box d-flex d-sm-none">
        <div class="media-body ">
          <h3 class="media-heading">Model zoo for genomics</h3>
        </div>
      </div>
    </div>
    <div class="row mb-5">
      <div class="col-md-6 col-sm-12">
        <img src="{{ url_for('static', filename='img/fig1_v8_hires.png')}}" class="img-fluid p-3" alt="Kipoi overview figure">
      </div>
      <div class="col-md-6 col-sm-12">
        <p class="p-3">
          Kipoi (pronounce: k√≠pi; from the Greek &kappa;&#x3AE;&pi;&omicron;&iota;: gardens) is an <a href='https://github.com/kipoi/kipoi'>API</a> and a <a href='https://github.com/kipoi/models'>repository</a> of ready-to-use
          trained models for genomics. It currently contains {{ n_models }}
          different models, covering canonical predictive tasks in transcriptional and post-transcriptional gene regulation.

          Kipoi's API is implemented as a python package (<a href='https://github.com/kipoi/kipoi'>github.com/kipoi/kipoi</a>) and it is also accessible from the command line or R.
          <!-- API to use, re-use and combine these models efficiently
              and a flexible toolbox for downstream analysis like variant effect prediction. -->

          <!-- The model is an abstraction over the model from major machine-learning frameworks in python
              (Keras, TensorFlow, PyTorch, Scikit-learn) with the option to write a custom model.
              The dataloader loads data from raw files and returns processed data, ready to be consumed by the
              model together with the metadata.
              This allows the models to be used directly with typical bioinformatics file formats like fasta or
              bed and to, out-of-the-box, predict the effects of genetic variants.
              The source is a remote storage for models and dataloaders. Users can access models from private or public sources.
              The main public Kipoi source is hosted on github.com/kipoi/models with model parameters being stored with Git Large File Storage (LFS).
              Model and dataloader are defined with YAML description files.
              These provide general information about the model, authors, supported downstream analysis and required dependencies managed by pip or conda.
              Models and dataloaders are contributed and reviewed through pull-requests on github.  -->

          <!-- - Unified and easy way to run models on new data
            - Model dependencies managed by Conda
            - Variant effect prediction for x models
            - Transparency and overview
            - Enriched functionality for the contributed models (less boilerplate code)
            - Flexibility and ease of contribution
          -->
        </p>
      </div>
    </div>

    <hr class="w-100 clearfix">

    <div class="row mb-5">
      <div class="col-lg-4 col-md-6 col-sm-6 col-12">
        <h4>Numbers</h4>
        <p><strong># of models:</strong> {{n_models}}</p>
        <p><strong># of model groups:</strong> {{n_groups}}</p>
        <p><strong># of contributors:</strong> {{n_contributors}}</p>
        <p><strong># of model groups supporting postprocessing:</strong></p>
        <ul>
          <li><strong>Variant effect prediction:</strong> {{ n_postproc_score_variants }}/{{ n_groups }}</li>
        </ul>
      </div>

      <!--Show some plots about Kipoi-->
      <div class="col-lg-4 col-md-5 col-sm-6 col-9">
        <h4>Model groups by framework</h4>
        <div style="max-width:85%; max-height:85%;" class="px-3">
          <canvas id="models-by-framework"></canvas>
        </div>
      </div>
      <!-- <div class="col-md-4">
        <h4>Model groups by <br />license</h4>
        <canvas id="models-by-license"></canvas>
        </div>
      -->
      <div class="col-lg-4 col-md-4 col-sm-4 col-9">
        <h4>Model groups by tag</h4>
        <div style="height: 280px">
          <canvas id="models-by-genomics-tag"></canvas>
        </div>
      </div>
    </div>
    
    <hr class="w-100 clearfix">
    
    <div class="row">
      <div class="col">
        <h3>Getting Started</h3>
      </div>
    </div>
    
    <div class="row mb-5">
      <div class="col-lg-4">
        <h4>Use models</h4>
        <ul>
          <!-- <li>Install git-lfs: <code>$ conda install -c conda-forge git-lfs && git lfs install</code></li> -->
          <li>Install Kipoi: <code>$ pip install kipoi</code></li>
          <li>Choose the model from <a href="{{ url_for('models.list_groups', group_name='') }}">http://kipoi.org/models</a></li>
          <li>Use the model from CLI/python/bash by copying code snippets from <a href="../models/Basenji">http://kipoi.org/models/my_model</a></li>
        </ul>
        <a class="btn btn-outline-info waves-effect" href="../docs/using/01_Getting_started/">Learn more</a>
      </div>

      <div class="col-lg-4">
        <h4>Contribute models</h4>
        <ul>
          <!-- <li>Install git-lfs: <code>$ conda install -c conda-forge git-lfs && git lfs install</code></li> -->
          <li>Install Kipoi: <code>$ pip install kipoi</code></li>
          <li>Run <code>$ kipoi ls</code></li>
          <li>Explore the models locally at <code>~/.kipoi/models</code></li>
          <li>Upload model weights to <a href='https://zenodo.org'>zenodo</a> or <a href='https://figshare.com/'>figshare</a>.</li>
          <li>Add the model to <code>~/.kipoi/models/</code> (copy&edit the existing model or run <code>kipoi init</code>)</li>
          <li>Test the model: <code>kipoi test ~/.kipoi/models/my_model</code></li>
          <li>Submit the pull-request to <a href="https://github.com/kipoi/models">github.com/kipoi/models</a></li>
        </ul>
        <a class="btn btn-outline-info waves-effect" href="../docs/contributing/01_Getting_started/">Learn more</a>
      </div>

      <div class="col-lg-4">
        <h4>Contribute code</h4>
        <ul>
          <li>Setup the development environment: <code>$ pip install kipoi[develop]</code></li>
          <li>Read <a href="https://github.com/kipoi/kipoi/blob/master/CONTRIBUTING.md">CONTRIBUTING.md</a></li>
          <li>Submit an issue/pull-request to <a href="https://github.com/kipoi/kipoi">github.com/kipoi/kipoi</a></li>
        </ul>
      </div>
    </div>

    <hr class="w-100 clearfix">

    <div class="row mb-4">
      <div class="col-lg-4 col-md-12">
          <div style="padding:10px"></div>
          {% include 'models/twitter.html' %}
      </div>
      
      <div class="col-lg-8 col-md-12">
        <!-- Some news -->  
        <!--<h4>News</h4>
          <ul>
            <li></li>
            <li></li>
            <li></li>
          </ul>-->
      </div>
    </div>
    
  </div>
</div>
{% endblock %}

{% block script %}
<script src="https://codepen.io/anon/pen/aWapBE.js"></script>
<script language="JavaScript">
  $(document).ready(function() {
    var color_pallete = 'cb-Set2';
    // models-by-framework
    var colors = palette(color_pallete, {{ models_by_framework_values | length }}).map(function(hex) {
      return '#' + hex;
    });
    var ctxP = document.getElementById("models-by-framework").getContext('2d');
    var myPieChart = new Chart(ctxP, {
      type: 'doughnut',
      data: {
        labels: {{ models_by_framework_keys | safe }},
        datasets: [
          {
            data: {{ models_by_framework_values }},
            backgroundColor: {{ models_by_framework_colors | safe }},
	    hoverBackgroundColor: {{ models_by_framework_colors | safe }},
            /* backgroundcolor: ["#F7464A", "#46BFBD", "#FDB45C", "#949FB1", "#4D5360"],
	       hoverBackgroundColor: ["#FF5A5E", "#5AD3D1", "#FFC870", "#A8B3C5", "#616774"]*/
          }
        ]
      },
      options: {
        responsive: true
      }    
    });
    // models-by-license
    /* var ctxP = document.getElementById("models-by-license").getContext('2d');
     * var myPieChart = new Chart(ctxP, {
     *   type: 'doughnut',
     *   data: {
     *     labels: {{ models_by_license_keys | safe }},
     *     datasets: [
     *       {
     *         data: {{ models_by_license_values }},
     *         backgroundColor: ["#F7464A", "#46BFBD", "#FDB45C", "#949FB1", "#4D5360"],
     *         hoverBackgroundColor: ["#FF5A5E", "#5AD3D1", "#FFC870", "#A8B3C5", "#616774"]
     *       }
     *     ]
     *   },
     *   options: {
     *     responsive: false
     *   }    
     * });*/
    //bar
    var ctxB = document.getElementById("models-by-genomics-tag").getContext('2d');
    var myBarChart = new Chart(ctxB, {
      type: 'horizontalBar',
      data: {
        labels: {{ models_by_genomics_tag_keys | safe}},
        datasets: [{
          label: null,
          data: {{ models_by_genomics_tag_values}},
          /* backgroundColor: ["#F7464A80", "#46BFBD80", "#FDB45C80", "#949FB180", "#F3E06980", "#4D536080", "#AAABAA80"],*/
	  backgroundColor: palette(color_pallete, {{ models_by_genomics_tag_values | length }}).map(function(hex) {
	    return '#' + hex;
	  }),
	  /* borderColor: [
	     'rgba(255,99,132,1)',
	     'rgba(54, 162, 235, 1)',
	     'rgba(255, 206, 86, 1)',
	     'rgba(75, 192, 192, 1)',
	     'rgba(153, 102, 255, 1)',
	     'rgba(255, 159, 64, 1)',
	     'rgba(255, 159, 64, 1)'
	     ],*/
          /* borderWidth: 1*/
        }]
      },
      options: {
        maintainAspectRatio: false,
	responsive: true,
	legend: {
	  display: false,
	},

        scales: {
	  ticks: {
	    autoSkip: false
	  },
	  xAxes: [{
	    beginAtZero: true,
	    ticks: {
	      beginAtZero: true,
	      autoSkip: false
	      /* maxRotation: 90*/
	    },
	  }],
          yAxes: [{
            beginAtZero:true,
            ticks: {
              beginAtZero:true
            }
          }]
        }
      }
    });
    
  });
</script>
{% endblock%}
