{% from "macros.html" import not_none, parse_cite_as, show_authors %}
{% extends 'base.html' %}
{% block content %}

<!-- - what is it?
- when is it?
- how to register?
- agenda?
- next speaker?
- past speakers
- apply as speaker-->
<div class="row" style="margin-right: 0; margin-left: 0;">
    <div class="col-lg-5 mx-auto">
        <!-- <h3 class="media-heading">About</h3> -->
        <h3>Kipoi - Seminar</h3>
        <p>
            The monthly virtual seminar series is designed as a platform for interested Kipoi users and developers and
            will host talks on the applications of deep learning on biological data.
            The seminar is held on every first Wednesday of the month at 5:30 p.m. - 6:30 p.m. CET. We are also happy to
            share the recordings of the seminar on <a href="https://www.youtube.com/channel/UCL_iKCHUxepOi7_Wk010--g">YouTube</a>.
        </p>
        <h4>How to take part</h4>
        <p>
            The Virtual Seminar Series takes place via Zoom.
            To take part in the seminar, you can register for the online Zoom conference <a
                href='https://tum-conf.zoom.us/meeting/register/u5MpcuCvqDwoHtRxJTdDig7co4LJoiHrYugn'>here</a>.
            Your personal join link will be valid for all upcoming lectures of the series.
        </p>
        <h4>How to apply as a speaker</h4>
        <p>The seminar is a great opportunity to present your recent work to a large international audience.
            If you want to apply as a speaker, please use the contact in the registration confirmation email.</p>
        <h4>Next seminar</h4>
        <h6> Title TBD: </h6> 2 June 2021 5:30 p.m. - 6:30 p.m. Central European Summer time

        <p>Speaker: <strong><a href = "https://www.seeliglab.org/">Johannes Linder</a></strong>, University of Washington, Seattle </p>
        <!--<strong>Abstract:</strong>
        <p align="justify">
            I will present (unpublished research) integrative analysis of single cell chromatin accessibility and expression data
            from early stages of human fetal heart development. We train and interpret deep learning models of base-resolution chromatin
            accessibility profiles to dissect cis-regulatory syntax of combinatorial TF binding in regulatory elements controlling
            dynamic gene expression and cell state transitions across all the major cellular lineages in human cardiogenesis.
            We predict the regulatory impact of de-novo non-coding mutations in congenital heart disease to prioritize putative
            causal mutations, transcription factors, regulatory elements and their target genes in putative causal cell types.
        </p>-->
<!--        <p align="justify" >-->
<!--            SAILER:-->
<!--            Single-cell sequencing assay for transposase-accessible chromatin (scATAC-seq) provides new opportunities to-->
<!--            dissect epigenomic heterogeneity and elucidate transcriptional regulatory mechanisms. However, computational-->
<!--            modelling of scATAC-seq data is challenging due to its high dimension, extreme sparsity, complex-->
<!--            dependencies, and high sensitivity to confounding factors from various sources.-->
<!--            In this work we propose a new deep generative model framework, named SAILER, for analyzing scATAC-seq data. </p>-->
<!--            <div class="collapse" id="collapseExample"> <p> SAILER aims to learn a low-dimensional nonlinear latent representation of each cell that defines its-->
<!--            intrinsic chromatin state, invariant to extrinsic confounding factors like read depth and batch effects.-->
<!--            SAILER adopts the conventional encoder-decoder framework to learn the latent representation but imposes-->
<!--            additional constraints to ensure the independence of the learned representations from the confounding-->
<!--            factors. Experimental results on both simulated and real scATAC-seq datasets demonstrate that SAILER learns-->
<!--            better and biologically more meaningful representations of cells than other methods. Its noise-free cell-->
<!--            embeddings bring in significant benefits in downstream analyses: Clustering and imputation based on SAILER-->
<!--            result in significant improvements over existing methods, respectively. Moreover, because no matrix-->
<!--            factorization is involved, SAILER can easily scale to process millions of cells. We implemented SAILER into-->
<!--            a software package, freely available to all for large-scale scATAC-seq data analysis.-->

<!--            UFold:-->
<!--            For many RNA molecules, the secondary structure is essential for the correct function of the RNA. Predicting-->
<!--            RNA secondary structure from nucleotide sequences is a long-standing problem in genomics, but the prediction-->
<!--            performance has reached a plateau over time. Traditional RNA secondary structure prediction algorithms are-->
<!--            primarily based on thermodynamic models through free energy minimization, which imposes strong prior-->
<!--            assumptions and is slow to run.-->
<!--            Here we propose a deep learning-based data-driven method, called UFold, for RNA secondary structure-->
<!--            prediction, trained directly on annotated data without any thermodynamic assumptions. UFold improves-->
<!--            substantially upon previous models. UFold is also fast with an inference time about 160ms per sequence up to-->
<!--            1600bp length. We provide an online web server that implements UFold for RNA structure prediction and is-->
<!--                made freely available.-->
<!--            </p> </div>-->
<!--            <button class="btn btn-outline-info waves-effect" type="button" color="" data-toggle="collapse" data-target="#collapseExample" aria-expanded="false" aria-controls="collapseExample">-->
<!--                Read More-->
<!--            </button>-->
        <h4>Upcoming speakers</h4>
        <div class="container-fluid">
        <ul class="list-unstyled">
            <li>7 July 2021 - <a href="http://zhanglab.net/">Qiangfeng Cliff Zhang</a>, Tsinghua University, Beijing</li>
            <li>4 August 2021 - <a href="https://www.mskcc.org/research/ski/labs/christina-leslie">Christina Leslie</a>, Sloan Kettering Institute, New York</li>
        </ul>
        </div>
        <h4>Previous speakers</h4>
        <div class="container-fluid">
        <ul class="list-unstyled">
            <li>5 May 2021 - <a href = "https://profiles.stanford.edu/anshul-kundaje">Anshul Kundaje</a>, Stanford University, Stanford</li>
            <li>7 April 2021 - <a href="https://cellfate.uci.edu/themes/center-fellows/">Yingxin Cao</a>, UC Irvine </li>
            <li>3 March 2021 - <a href="https://stanford.edu/~avanti/">Avanti Shrikumar</a>, Stanford University, Stanford  </li>
            <li>3 February 2021 - <a href = "https://www.mdc-berlin.de/ohler">Uwe Ohler</a>, Max-Delbr√ºck-Center for Molecular Medicine, Berlin</li>
            <li>2 December 2021 - <a href="https://www.rdm.ox.ac.uk/people/ron-schwessinger">Ron Schwessinger</a>, Radcliffe Department of Medicine, Oxford</li>
            <li>4 November 2020 - <a href="http://www.davidrkelley.com/">David Kelley</a>, Calico, San Francisco</li>
            <li>7 October 2020 - <a href="https://www.calicolabs.com/">Vikram Agarwal</a>, Calico, San Francisco</li>
        </ul>
        </div>
        </div>
    </div>
    {% endblock %}